fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -279,13 +281,17 @@ void FixedLagSmoother::optimizerTimerCallback(const ros::TimerEvent& event)
     {
       std::lock_guard<std::mutex> lock(optimization_requested_mutex_);
       optimization_request_ = true;
-      optimization_deadline_ = event.current_expected + params_.optimization_period;
+      // TODO(CH3): ??????? No idea what this does...
+      // XXX event.current_expected refers to when ROS planned to execute the callback, not the current time.
+      //     wall-time and measurement-time should be decoupled further
+      // optimization_deadline_ = event.current_expected + params_.optimization_period;
+      optimization_deadline_ = get_clock()->now() + params_.optimization_period;
```

Comment:
The documentation says updates should happen at a fixed frequency.

https://github.com/locusrobotics/fuse/blob/1118d16856017a8b8430c0d24de66e1c993d5d43/fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother.h#L70-L7

I think using `now() + period` would shift when the next update will happen. Maybe `now()` + [`timer->time_until_trigger()`](https://github.com/ros2/rclcpp/blob/7c6785176a878cf16eb98995b732f405b9e776ed/rclcpp/include/rclcpp/timer.hpp#L121-L122) would have the same behavior as before?

Reply:
https://github.com/locusrobotics/fuse/pull/283/commits/59d10ddc448f2a21928d5104dabe494e0e37f8e4